stages/org.osbuild.systemd-journald

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python3
+"""
+Configures systemd-journald
+
+The 'config' option allows to create a systemd-journald configuration drop-in
+file in `/usr/lib/systemd/journald.conf.d` with the name `filename`.
+
+Drop-in configuration files can specify the following subset of options:
+- 'Journal' section
+  -  'Storage' option
+  -  'Compress' option
+  -  'SplitMode' option
+  -  'MaxFileSec' option
+  -  'MaxRetentionSec' option
+  -  'SyncIntervalSec' option
+  -  'Audit' option
+
+At least one option must be specified in the 'Journal' section.
+"""
+
+import configparser
+import os
+import sys
+
+import osbuild.api
+
+SCHEMA = r"""
+"additionalProperties": false,
+"properties": {
+    "filename": {
+        "type": "string",
+        "description": "Name of the systemd-journald drop-in file.",
+        "pattern": "^([[a-zA-Z0-9_]|\\.|-]){1,250}\\.conf$"
+    },
+    "config": {
+        "additionalProperties": false,
+        "type": "object",
+        "description": "systemd-journald configuration.",
+        "minProperties": 1,
+        "properties": {
+            "Journal": {
+                "additionalProperties": false,
+                "type": "object",
+                "description": "'Journal' configuration section.",
+                "minProperties": 1,
+                "properties": {
+                    "Storage": {
+                        "type": "string",
+                        "description": "Controls where to store journal data.",
+                        "pattern": "^(volatile|persistent|auto|none)$"
+                    },
+                    "Compress": {
+                        "type": "string",
+                        "description": "Sets whether the data objects stored in the journal should be compressed or not. Can also take threshold values.",
+                        "pattern": "^(yes|no|[0-9]+(K|M|G))$"
```

Comment:
I would drop the pattern, since systemd allows mixing multiple modifiers consequetively and I don't think we want to match on that.

Reply:
According to the docs, as I understood them, that pattern should match what is allowed. What do you mean by multiple modifiers?

Reply:
Something like `5G 400M 10K` to mean `5.4001G`, but "base-1024". I haven't checked whether this key allows it, but the generic parsers in systemd allow this.

Reply:
I see, yep, then the pattern doesn't match that at all. Will drop the pattern in this field and the others that have the same issue :+1: